### PR TITLE
MHV-55309-Add-Refill-Logic

### PR DESCRIPTION
**Pull request by agravell047**
## Summary

- Added logic to refill multiple prescriptions
- Added logic to retrieve a list of only refillable/renewable prescriptions
- Added test for new endpoint

## Related issue(s)

https://jira.devops.va.gov/browse/MHV-55039
> As a Medications developer, I want to create a new subpage for refilling so that it matches mocks

## Testing done

- Tested locally
- Ran tests locally
- Added test

## What areas of the site does it impact?
MHV Medications App

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by agravell047** [comment]
I'm not sure if the rubocop findings can be fixed by using `.zero?`. When trying this I see: `undefined method `zero?' for nil:NilClass`

**Review by vmenshutin-bylight — APPROVED** [review]

File: modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb [hunk 1]
```diff
@@ -47,6 +47,29 @@ def refill
...
       end
 
+      def refill_prescriptions
+        ids = params[:ids]
+        begin
+          ids.each do |id|
+            client.post_refill_rx(id)
+          end
+        rescue
+          puts "Error refilling prescription: #{e.message}"
```
Comment (kpethtel) [inline modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb:57]: Do you not have to explicitly `rescue => e` in order to access the exception object? I don't think that used to work. I wonder when that changed.

File: modules/my_health/spec/request/v1/prescriptions_request_spec.rb [hunk 2]
```diff
@@ -130,6 +130,18 @@
         expect(JSON.parse(response.body)['meta']['pagination']['per_page']).to eq(10)
       end
 
+      it 'responds to GET #refill_list with list of refillable prescriptions' do
```
Comment (kpethtel) [inline modules/my_health/spec/request/v1/prescriptions_request_spec.rb:133]: I think you may have changed the method name away from `refill_list` at some point.

**Review by kpethtel — APPROVED** [review]
I assume the mobile team is a code owner on this, and it's fine as far as mobile is concerned. I think it would be better if `refill_prescriptions` were tested.

**Review by robertbylight — APPROVED** [review]
lgtm

File: modules/my_health/spec/request/v1/prescriptions_request_spec.rb (hunk 2 above)
Comment (agravell047) [reply modules/my_health/spec/request/v1/prescriptions_request_spec.rb:133]: @kpethtel I will fix this up, thanks for pointing it out.

File: modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb (hunk 1 above)
Comment (agravell047) [reply modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb:57]: As far as I know, this hasn't changed but I'm not sure why it didn't throw an error. I'm surprised my linter didn't catch it either.

**Review by rjohnson2011 — APPROVED** [review]

**Pull request merged**